cli/command/service/update.go

```diff
@@ -934,3 +943,64 @@ func updateHealthcheck(flags *pflag.FlagSet, containerSpec *swarm.ContainerSpec)
 	}
 	return nil
 }
+
+type byNetworkTarget []swarm.NetworkAttachmentConfig
+
+func (m byNetworkTarget) Len() int      { return len(m) }
+func (m byNetworkTarget) Swap(i, j int) { m[i], m[j] = m[j], m[i] }
+func (m byNetworkTarget) Less(i, j int) bool {
+	return m[i].Target < m[j].Target
+}
+
+func updateNetworks(apiClient client.NetworkAPIClient, flags *pflag.FlagSet, spec *swarm.ServiceSpec) error {
+	ctx := context.Background()
```

Comment:
It's necessary for this function to make API calls. Even if translation from network names to network IDs is done in the engine, as you prefer, it is done before the service is stored. When the service is retrieved from the manager `NetworkAttachmentConfig.Target` will contain an ID. If a user tries to remove a network by name, the name needs to be resolved to an ID so that the appropriate `NetworkAttachmentConfig` can be removed.